src/ab-test-preview.js

```diff
@@ -2,58 +2,79 @@ import { getLetter } from '../src/utils';
 
 const { __, sprintf } = wp.i18n;
 
-document.querySelectorAll( '.ab-test-xb-preview' ).forEach( xb => {
+/**
+ * Create the preview tabs
+ */
+function createTabbedPreviews() {
+	document.querySelectorAll( '.ab-test-xb-preview' ).forEach( xb => {
 
-	const xbPostId = xb.dataset.postId;
-	const templates = document.querySelectorAll( 'template[data-parent-id="' + xb.dataset.clientId + '"]' );
-	const tabContainer = xb.querySelector( '.ab-test-xb-preview__tabs' );
-	const tabContent = xb.querySelector( '.ab-test-xb-preview__content' );
+		const xbPostId = xb.dataset.postId;
+		const templates = document.querySelectorAll( 'template[data-parent-id="' + xb.dataset.clientId + '"]' );
+		const tabContainer = xb.querySelector( '.ab-test-xb-preview__tabs' );
+		const tabContent = xb.querySelector( '.ab-test-xb-preview__content' );
```

Comment:
I can't see where we check if the tabs are already set up, I was thinking something like
```suggestion
		const tabContent = xb.querySelector( '.ab-test-xb-preview__content' );

		// Return if we've processed this XB already.
		if ( tabContainer.children.length > 0 ) {
			return;
		}
```

Reply:
I had done this slightly differently further done in the function, but this method makes more sense.